### ISSUE TITLE
Make the global content repo site private by default

### ISSUE THREAD

**Issue by roborourke** [humanmade/altis-core#261]
The content repository should be for internal use by default, with the option to allow for authenticated API requests. We also want to keep the ability to turn this off and make the site public in the future, so this should not be enforced to be private.

(While it's possible to also edit the URL when changing the public flag, any issues with that are not in scope for this ticket.)

Acceptance criteria:
* [x] As a developer, when I view the documentation about the global content repo, I should see that the site is private by default
* [x] When the global content repo is created, it should be marked as private and require login
* [x] As a user, when I edit the site in the network admin, I am able to make the content repo site public

**Comment by rmccue** [comment]
In #259, we switched to using authenticated REST API requests internally, which unblocked this issue.

**Comment by roborourke** [comment]
Fixed by #263
